clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -3167,6 +3167,15 @@ Parser::ParseCXXClassMemberDeclaration(AccessSpecifier AS,
 
     DeclaratorInfo.complete(ThisDecl);
 
+    if (TemplateInfo.Kind != ParsedTemplateInfo::NonTemplate) {
+      if (Tok.is(tok::comma)) {
+        Diag(Tok, diag::err_multiple_template_declarators)
+            << (int)TemplateInfo.Kind;
+        SkipUntil(tok::semi, StopBeforeMatch);
+      }
+      break;
+    }
```

Comment:
I am not sure it make sense to do that in the parser.
Can we maybe to that in `FinalizeDeclaratorGroup`?
At that point we should have fully formed `Decl`s and we can check whether they are templates

Reply:
@cor3ntin We diagnose namespace scope template/explicit specialization/explicit instantiation declarations with multiple declarators [in the parser as well](https://github.com/llvm/llvm-project/blob/c20811b659766469e80317702ee204ef6c4375b4/clang/lib/Parse/ParseTemplate.cpp#L370)... is there a reason not to?

Reply:
How about issuing the diagnostics in the parser, but instead of skipping to the semicolon we just continue parsing? That way the diagnostics are issued in lexical order when a declaration with multiple errors is encountered.